packages/components/modal/src/modal.js

```diff
@@ -81,5 +84,29 @@ customElementDefineHelper(
 				<ts-cover class="ts-dialog-cover" ?data-visible=${this.visible} @click="${this.close}"></ts-cover>
 			`;
 		}
+
+		firstUpdated() {
+			document.addEventListener('keydown', this.onKeyDown);
+		}
+
+		disconnectedCallback() {
+			document.removeEventListener('keydown', this.onKeyDown);
+			super.disconnectedCallback();
+		}
+
+		onKeyDown(event) {
```

Comment:
Maybe for now, we can atleas move this method somewhere and reuse it as event listener call back. For now a helper function would be fine I guess. It can get parameters close event and condition. What do you think?